### DISCUSSION TITLE
Configure no-underscore-dangle to allow function names, not in classes

### DISCUSSION THREAD

**Discussion by robopzet** [eslint/eslint#16108] [Q&A]
What combination of options is required to configure the rule **no-underscore-dangle** to:

1. flag underscores in all class method names, parameters and fields.
2. allow underscores in the names of standalone functions. Ideally only allow one as prefix, but I don't think the rule can be that specific.

I've seen options for function parameters and class methods, but not functions.

**Comment by sam3k** [comment]
Hi @robopzet, Just [tried](https://eslint.org/play/#eyJ0ZXh0IjoiIC8qIGVzbGludCBxdW90ZXM6IFtcImVycm9yXCIsIFwic2luZ2xlXCJdICovXG5jb25zdCBhID0gJ2InO1xuXG5jb25zdCBmb28gPSAoKSA9PiB7XG4gIHJldHVybiBhO1xufVxuXG5mb28oKTtcblxuY29uc3QgX2JvbyA9ICdoZWxsbyc7XG5cbmNvbnN0IF9mb28gPSAoKSA9PiB7XG4gIHJldHVybiBfYm9vO1xufVxuXG5jbGFzcyBQZXJzb24ge1xuICBfam9iVGl0bGUgPSAnJ1xuXG4gIF9zb21lUHJpdmF0ZU1ldGhvZCgpIHtcbiAgICByZXR1cm4gdHJ1ZTtcbiAgfVxuICBcbiAgY29uc3RydWN0b3IoZmlyc3ROYW1lLCBsYXN0TmFtZSkge1xuICAgIHRoaXMubmFtZSA9IGAke2ZpcnN0TmFtZX0gJHtsYXN0TmFtZX1gO1xuXG4gICAgdGhpcy5fc29tZVByaXZhdGVNZXRob2QoKTtcbiAgfVxufVxuXG5uZXcgUGVyc29uKCdKYW5lJywgJ1NtaXRoJyk7XG5cblxuX2ZvbygpOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOlsiZXJyb3IiXSwiZm9yLWRpcmVjdGlvbiI6WyJlcnJvciJdLCJnZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOlsiZXJyb3IiXSwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOlsiZXJyb3IiXSwibm8tY2xhc3MtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbXBhcmUtbmVnLXplcm8iOlsiZXJyb3IiXSwibm8tY29uZC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3QtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6WyJlcnJvciJdLCJuby1jb250cm9sLXJlZ2V4IjpbImVycm9yIl0sIm5vLWRlYnVnZ2VyIjpbImVycm9yIl0sIm5vLWRlbGV0ZS12YXIiOlsiZXJyb3IiXSwibm8tZHVwZS1hcmdzIjpbImVycm9yIl0sIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6WyJlcnJvciJdLCJuby1kdXBlLWVsc2UtaWYiOlsiZXJyb3IiXSwibm8tZHVwZS1rZXlzIjpbImVycm9yIl0sIm5vLWR1cGxpY2F0ZS1jYXNlIjpbImVycm9yIl0sIm5vLWVtcHR5IjpbImVycm9yIl0sIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1lbXB0eS1wYXR0ZXJuIjpbImVycm9yIl0sIm5vLWV4LWFzc2lnbiI6WyJlcnJvciJdLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOlsiZXJyb3IiXSwibm8tZXh0cmEtc2VtaSI6WyJlcnJvciJdLCJuby1mYWxsdGhyb3VnaCI6WyJlcnJvciJdLCJuby1mdW5jLWFzc2lnbiI6WyJlcnJvciJdLCJuby1nbG9iYWwtYXNzaWduIjpbImVycm9yIl0sIm5vLWltcG9ydC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW5uZXItZGVjbGFyYXRpb25zIjpbImVycm9yIl0sIm5vLWludmFsaWQtcmVnZXhwIjpbImVycm9yIl0sIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjpbImVycm9yIl0sIm5vLWxvc3Mtb2YtcHJlY2lzaW9uIjpbImVycm9yIl0sIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjpbImVycm9yIl0sIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6WyJlcnJvciJdLCJuby1uZXctc3ltYm9sIjpbImVycm9yIl0sIm5vLW5vbm9jdGFsLWRlY2ltYWwtZXNjYXBlIjpbImVycm9yIl0sIm5vLW9iai1jYWxscyI6WyJlcnJvciJdLCJuby1vY3RhbCI6WyJlcnJvciJdLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOlsiZXJyb3IiXSwibm8tcmVkZWNsYXJlIjpbImVycm9yIl0sIm5vLXJlZ2V4LXNwYWNlcyI6WyJlcnJvciJdLCJuby1zZWxmLWFzc2lnbiI6WyJlcnJvciJdLCJuby1zZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjpbImVycm9yIl0sIm5vLXNwYXJzZS1hcnJheXMiOlsiZXJyb3IiXSwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOlsiZXJyb3IiXSwibm8tdW5kZWYiOlsiZXJyb3IiXSwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOlsiZXJyb3IiXSwibm8tdW5yZWFjaGFibGUiOlsiZXJyb3IiXSwibm8tdW5zYWZlLWZpbmFsbHkiOlsiZXJyb3IiXSwibm8tdW5zYWZlLW5lZ2F0aW9uIjpbImVycm9yIl0sIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6WyJlcnJvciJdLCJuby11bnVzZWQtbGFiZWxzIjpbImVycm9yIl0sIm5vLXVudXNlZC12YXJzIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6WyJlcnJvciJdLCJuby11c2VsZXNzLWNhdGNoIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtZXNjYXBlIjpbImVycm9yIl0sIm5vLXdpdGgiOlsiZXJyb3IiXSwicmVxdWlyZS15aWVsZCI6WyJlcnJvciJdLCJ1c2UtaXNuYW4iOlsiZXJyb3IiXSwidmFsaWQtdHlwZW9mIjpbImVycm9yIl0sIm5vLXVuZGVyc2NvcmUtZGFuZ2xlIjpbImVycm9yIix7ImVuZm9yY2VJbk1ldGhvZE5hbWVzIjpmYWxzZSwiZW5mb3JjZUluQ2xhc3NGaWVsZHMiOmZhbHNlLCJhbGxvd0FmdGVyVGhpcyI6dHJ1ZSwiYWxsb3dBZnRlclN1cGVyIjpmYWxzZSwiYWxsb3dBZnRlclRoaXNDb25zdHJ1Y3RvciI6ZmFsc2UsImFsbG93RnVuY3Rpb25QYXJhbXMiOnRydWUsImFsbG93SW5BcnJheURlc3RydWN0dXJpbmciOnRydWUsImFsbG93SW5PYmplY3REZXN0cnVjdHVyaW5nIjp0cnVlfV19LCJlbnYiOnsiZXM2Ijp0cnVlfX19) doing a quick test to see if I can get this to work but there [doesn't seem to be a rule](https://eslint.org/docs/latest/rules/no-underscore-dangle) that allows for this. If this is only going to be used in a handful of functions; you could try to add them to the `allow` object.
